tf_agents/agents/dqn/examples/v2/train_eval.py

```diff
@@ -87,6 +89,9 @@ def train_eval(
     train_sequence_length=1,
     # Params for QNetwork
```

Comment:
Instead of passing all the different parameters for all the different types of q_networks, instead pass the q_network_fn(observation_spec, action_spec) that builds the appropriate network, ex. q_network, duel_q_network, create_recurrent_network or create_feedforward_network

Reply:
I agree. However, the q_network and duel_q_network functions takes different number of arguments. In my understanding, we could use network_fn(**kwargs)?

Reply:
I suggest we consolidate the first 2 arguments across all the networks to be `observation_spec, action_spec` and then each one may have their specific extra kwargs.

Reply:
So you mean changing the prototype of QNetwork class to accept variable number of arguments? Please correct me if I am not right. If that is the case, that change has broader impact and I would suggest doing it in a separate code refactoring task.

Reply:
I have consolidated the q_network fns to a function q_network_fn.: [ab7bee2](https://github.com/tensorflow/agents/pull/551/commits/ab7bee286ead30d7c5d4353627f9da90b6f0c316)